config/locales/translation_ja.yml

```
@@ -0,0 +1,28 @@
+ja:
+  activerecord:
+    models:
+      book: 本  #g
+
+    attributes:
+      book:
+        author: 著者  #g
+        memo: メモ  #g
+        picture: 写真  #g
+        title: タイトル  #g
+
+  helper:
+    common:
+      created: "%{name}を作成しました"
+      updated: "%{name}を更新しました"
+      destroyed: "%{name}を削除しました"
+
+  views:
+    common:
+      back: "%{name}の一覧に戻る"
+      edit: "%{name}を編集"
+      show: "%{name}の詳細を表示"
+      delete: "%{name}を削除"
+      create: "%{name}を新規作成"
+    books:
+      index:
+        title: 本一覧
```

これだけ

```suggestion
        title: %{name}一覧
```

にしなかった理由ってありますか？

---

タイトルという役割？的に「{何か}の一覧」というよりは、固有名詞かと思いハードコードで残した次第です。
パラメーターにするなら`list: %{name}一覧`の方かなとも一瞬思いましたが、気が変わってh1を「本のメモ一覧」とかする可能性もあるかなと思うと、固有のものとして残しておいた方がデザイン的なところで後々良いかと思いました。

---

うーん、「決め」の問題なので、絶対的な正解があるわけではないのですが、app/views/books/index.html.erbのh1タグだけは、タイトルと呼び、

```erb
<h1><%= t('views.books.index.title') %></h1>
```

app/views/books/new.html.erb のh1タグはタイトルと呼ばない理由ってあるんですかねえ？

```erb
<h1><%= t('views.common.create', name: Book.model_name.human) %></h1>
```

どっちも同じh1タグだし、↑も新規作成画面のタイトルと呼んでもいい気がするんですよね。

index.html.erbだけ特別扱いしているのが一貫性がない（他の人にとってその違いがわかりにくい）んじゃないかなーと思います。

> タイトルという役割？的に「{何か}の一覧」というよりは

"%{name}の一覧に戻る"という翻訳もあるので、「本の一覧」と呼んでも問題ない気がします。

> 気が変わってh1を「本のメモ一覧」とかする可能性もあるかな

そういう可能性についてはindex.html.erb以外の画面も同じ話なので、やはりindex.html.erbだけ特別扱いするのは不自然な気がしますね〜。

つまり、僕としては他のページと同じ対応で、

```yaml
  views:
    common:
      index: "%{name}の一覧"
```

でええんとちゃう？と思うわけです。